flow-server/src/main/java/com/vaadin/flow/router/internal/AbstractRouteRegistry.java

```diff
@@ -307,6 +306,11 @@ private void populateRegisteredRoutes(ConfiguredRoutes configuration,
         List<Class<? extends RouterLayout>> parentLayouts = getParentLayouts(
                 configuration, template);
 
+        var parameters = configuration.getParameters(template);
+        // exclude route from the menu if it has any required parameters
+        boolean excludeFromMenu = parameters != null && !parameters.isEmpty()
+                && parameters.values().stream()
+                        .anyMatch(param -> !param.getTemplate().contains("?"));
```

Comment:
```suggestion
                && parameters.values().stream().map(ParameterInfo::new).
                        .anyMatch(param -> !param.isOptional() && !param.isVarargs());
```
Wildcard routes should by definition work with no parameters.
Also we should use the ParameterInfo class instead of a magic char.